File: go/api/Cradle.go
package api

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Cradle 对应 Cradle.py。
type Cradle struct {
	*session.ApiGroup
}

// NewCradle 创建 Cradle API 分组。
func NewCradle(s *session.Session) *Cradle {
	return &Cradle{ApiGroup: session.NewApiGroup(s)}
}

// StatusInfo 对应 cradle/status-info。
func (c *Cradle) StatusInfo() (map[string]interface{}, error) {
	return c.S.Get("cradle/status-info", nil, "api")
}

// FeatureSwitch 对应 cradle/feature-switch。
func (c *Cradle) FeatureSwitch() (map[string]interface{}, error) {
	return c.S.Get("cradle/feature-switch", nil, "api")
}

// BasicInfo 对应 cradle/basic-info。
func (c *Cradle) BasicInfo() (map[string]interface{}, error) {
	return c.S.Get("cradle/basic-info", nil, "api")
}

// FactoryMac 对应 cradle/factory-mac。
func (c *Cradle) FactoryMac() (map[string]interface{}, error) {
	return c.S.Get("cradle/factory-mac", nil, "api")
}

// MacInfo 对应 cradle/mac-info。
func (c *Cradle) MacInfo() (map[string]interface{}, error) {
	return c.S.Get("cradle/mac-info", nil, "api")
}

File: src/api/cradle.rs
//! Cradle API group (`api/Cradle.py`).

use serde_json::Value;

use crate::connection::Connection;
use crate::errors::Result;

use super::get_value;

/// Cradle API group.
pub struct Cradle<'a> {
    conn: &'a Connection,
}

impl<'a> Cradle<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        Cradle { conn }
    }

    /// `cradle/status-info`. Cradle status information.
    pub fn status_info(&self) -> Result<Value> {
        get_value(self.conn, "api/cradle/status-info")
    }

    /// `cradle/feature-switch`. Cradle feature switch status.
    pub fn feature_switch(&self) -> Result<Value> {
        get_value(self.conn, "api/cradle/feature-switch")
    }

    /// `cradle/basic-info`. Cradle basic information.
    pub fn basic_info(&self) -> Result<Value> {
        get_value(self.conn, "api/cradle/basic-info")
    }

    /// `cradle/factory-mac`. Cradle factory MAC address.
    pub fn factory_mac(&self) -> Result<Value> {
        get_value(self.conn, "api/cradle/factory-mac")
    }

    /// `cradle/mac-info`. Cradle MAC address information.
    pub fn mac_info(&self) -> Result<Value> {
        get_value(self.conn, "api/cradle/mac-info")
    }
}
